scripts/tasks/execute-basic-buy.task.ts

```diff
@@ -0,0 +1,291 @@
+    .addFlag('debug', 'Debug mode')
+    .setAction(async (args: BasicBuyArgs, hre) => {
+        const { name: network } = hre.network
+        console.log(
+            `Network: ${network}. Using addresses from ${coalesceNetwork(args.forked || (network as Network))}\n`,
+        )
+        const hardhatUtils = new HardhatUtils(hre, args.forked)
+        const { addresses } = hardhatUtils
+        const startBlocks = getStartBlocksFor(args.forked || hre.network.name)
+
+        const { automationBot, automationExecutor } = await hardhatUtils.getDefaultSystem()
+
+        const events = await hre.ethers.provider.getLogs({
+            address: addresses.AUTOMATION_BOT,
+            topics: [automationBot.interface.getEventTopic('TriggerAdded'), bignumberToTopic(args.trigger)],
+            fromBlock: startBlocks.AUTOMATION_BOT,
+        })
+
+        if (events.length !== 1) {
+            throw new Error(
+                `Error looking up events. Expected to find a single TriggerAdded Event. Received: ${events.length}`,
+            )
+        }
+
+        const [event] = events
+        const { commandAddress, triggerData /* cdpId */ } = automationBot.interface.decodeEventLog(
+            'TriggerAdded',
+            event.data,
+            event.topics,
+        )
+        const {
+            vaultId,
+            type: triggerType,
+            executionCollRatio,
+            targetCollRatio,
+            maxBuyPrice,
+            continuous,
+            deviation,
+        } = decodeBasicBuyData(triggerData)
+        const info = [
+            `Command Address: ${commandAddress}`,
+            `Vault ID: ${vaultId.toString()}`,
+            `Trigger Type: ${triggerType.toString()}`,
+            `Execution Ratio: ${executionCollRatio.shiftedBy(-2).toFixed()}%`,
+            `Target Ratio: ${targetCollRatio.shiftedBy(-2).toFixed()}%`,
+            `Max Buy Price: ${maxBuyPrice.shiftedBy(-18).toFixed(2)}`,
+            `Continuous: ${continuous}`,
+            `Deviation: ${deviation.shiftedBy(-4).toFixed()}%`,
+        ]
+        console.log(`Found Trigger:\n\t${info.join('\n\t')}`)
+
+        if (!triggerType.eq(3)) {
+            throw new Error(`Trigger type \`${triggerType.toString()}\` is not supported`)
+        }
+
+        const executorSigner = await hardhatUtils.getValidExecutionCallerOrOwner(
+            automationExecutor,
+            hre.ethers.provider.getSigner(0),
+        )
+        const serviceRegistry = {
+            ...hardhatUtils.mpaServiceRegistry(),
+            feeRecepient:
+                network === Network.MAINNET
+                    ? '0xC7b548AD9Cf38721810246C079b2d8083aba8909'
+                    : await executorSigner.getAddress(),
+            exchange: addresses.EXCHANGE,
+        }
+        const { exchangeData, cdpData } = await getExecutionData(
+            hardhatUtils,
+            vaultId,
+            targetCollRatio,
```

Comment:
don't we need something more sophisticated here ? using deviation here in case targetCollRatio is near liquidationRatio ?